tests/integration/cluster.go

```diff
@@ -153,6 +159,10 @@ type ClusterConfig struct {
 
 	// UseIP is true to use only IP for gRPC requests.
 	UseIP bool
+	// UseBridge adds bridge between client and grpc server. Should be used in tests that
+	// want to manipulate connection or require connection not breaking despite server stop/restart.
+	UseBridge bool
```

Comment:
As this was previous default and majority of tests got to be changed to get `True` supplied here explicitly, 
have you considered making it `WithoutBridge` or `NoBridge` setting  to let opt-in for the new behaviour ?

Reply:
Using `WithoutBridge` would require less changes/work, but I don't think it would be better. I think that enabling bridge is very bad default, it is not as transparent as it might seem. It changes how broken connections are handled hiding some issues which we would want to fix. It also forces integration tests to go through unix socket, which is not officially supported (you cannot bind etcd server to unix socket). This makes maintaining integration and e2e tests much harder, as it increases differences between two environments. When something breaks you don't know if this is specific to integration test environment or etcd logic itself. 

I think we should strive to have integration and e2e tests be almost identical in setup, differing only in what is executed underneath (etcdctl binary in e2e and client library in integration). This way it is easy to attribute failures to etcd or environment. Using bridges as a common denominator creates unnecessary complexity, causing us to waste time on fixing  integration test code instead of focusing on etcd itself.

Reply:
SG. 

Does it mean, there is a follow-up TODO to check which of the existing tests would pass with 'False' ?

Reply:
I have only passed "true" only in tests that were failing, so I think this should be done. Only think it that there might be difference between what test was intended to test and it's implementation. Possibly if we changed implementation of some tests we could further reduce usage of bridge, but I don't think profit from this would outweigh amount of work that this would require us to invest.

I think it would make sense to propose rewrite of e2e/integration tests to unified framework and slowly migrate existing tests there, case by case. Removing bridge could be done at the same time and results would be more beneficial (halved amount of easier to debug tests)1 .